src/cryptography/x509.py

```diff
@@ -1344,3 +1344,52 @@ def extensions(self):
         """
         Returns an Extensions object containing a list of Revoked extensions.
         """
+
+
+class CertificateSigningRequestBuilder(object):
+    def __init__(self, version=Version.v1, subject_name=None, extensions=[]):
+        """
+        Creates an empty X.509 certificate request (v1).
+        """
+        self._version = Version.v1
+        self._subject_name = subject_name
+        self._extensions = extensions[:]
+
+    def set_version(self, version):
```

Comment:
Unlike certificates, version 1 is the only version of x509 CSRs that is officially defined. We raise errors on parsing certificates that have versions not defined by the RFC so to be consistent we should probably just remove `set_version` and force it to version 1.

That said, looking at our vectors we have quite a few custom CSRs that are defined as 0x2. That is likely an artifact of the way I generated them? Bears further investigation.